force-app/main/default/classes/ExchangeRatesController.cls

```
@@ -0,0 +1,335 @@
+     */
+    public void scheduleDailyUpdate() {
+        try {
+            String jobId = ExchangeRateScheduler.scheduleDailyUpdate();
+            this.statusMessage = 'Daily exchange rate update scheduled successfully. Job ID: ' + jobId;
+            setStatusMessage(STATUS_SUCCESS);
+            
+            // Reload scheduler info
+            loadSchedulerInfo();
+            
+        } catch (Exception e) {
+            this.statusMessage = 'Error scheduling daily update: ' + e.getMessage();
+            setStatusMessage(STATUS_ERROR);
+            System.debug(LoggingLevel.ERROR, 'Error in scheduleDailyUpdate: ' + e.getMessage());
+        }
+    }
+    
+    /**
+     * @description Schedules weekday exchange rate updates
+     */
+    public void scheduleWeekdayUpdate() {
+        try {
+            String jobId = ExchangeRateScheduler.scheduleWeekdayUpdate();
+            this.statusMessage = 'Weekday exchange rate update scheduled successfully. Job ID: ' + jobId;
+            setStatusMessage(STATUS_SUCCESS);
+            
+            // Reload scheduler info
+            loadSchedulerInfo();
+            
+        } catch (Exception e) {
+            this.statusMessage = 'Error scheduling weekday update: ' + e.getMessage();
+            setStatusMessage(STATUS_ERROR);
+            System.debug(LoggingLevel.ERROR, 'Error in scheduleWeekdayUpdate: ' + e.getMessage());
+        }
+    }
+    
+    /**
+     * @description Gets exchange rates for a specific currency
+     * @param currencyCode Currency code to filter by
+     * @return List of exchange rates for the currency
+     */
+    public List<Exchange_Rate__c> getExchangeRatesForCurrency(String currencyCode) {
+        if (String.isBlank(currencyCode)) {
+            return new List<Exchange_Rate__c>();
+        }
+        
+        try {
+            return [
+                SELECT Id, Currency_Code__c, Exchange_Rate__c, Amount__c, Rate_Date__c, Last_Updated__c,
```

🛑 **[Security] Query Must Enforce Field-Level Security (FLS)**

Since this is a `public` method that could be called from anywhere, we can't assume the running user has access to all the fields in this query. If a user is missing FLS on even one of these fields, the query will throw an exception and crash.

We should add the `WITH SECURITY_ENFORCED` clause to this SOQL query. It's the modern best practice for handling FLS in Apex and ensures the query is secure by default. It makes the code fail fast and predictably if permissions are missing, which is exactly what we want for a robust method.